src/logger/lwt_log_core.ml

```diff
@@ -47,6 +47,17 @@ let string_of_level = function
   | Error -> "error"
   | Fatal -> "fatal"
 
+let level_of_string str =
+  let str = (String.lowercase [@ocaml.warning "-3"]) str in
+  match str with
+  | "debug" -> Ok Debug
+  | "info" -> Ok Info
+  | "notice" -> Ok Notice
+  | "warning" -> Ok Warning
+  | "error" -> Ok Error
+  | "fatal" -> Ok Fatal
+  | _ -> Pervasives.Error (Printf.sprintf "invalid log level (%s)" str)
```

Comment:
I prefer `option` here :) But we already have the dependency on `result` (see [`lwt.opam`](https://github.com/ocsigen/lwt/blob/8d649540731590bea9e3d2f4ed9fb01af2d508ee/lwt.opam#L37)).